libs/content-search-toolkit/src/types/shared.ts

```diff
@@ -11,11 +11,17 @@ export enum SortField {
   POPULAR = 'popular',
 }
 
-export type elasticTagField = {
-  key: string
-  type: string
-  value?: string
-}
+export type elasticTagField =
+  | {
+      key?: string
+      type: string
+      value: string
+    }
+  | {
+      key: string
+      type: string
+      value?: string
+    }
```

Comment:
Have you considered updating the mapper so the `value` is written to the `key`?
This would probably allow for the desired functionality without needing this change